monai/networks/nets/efficientnet.py

```diff
@@ -0,0 +1,805 @@
+        out_channels = _round_filters(32, width_coefficient, depth_divisor)  # number of output channels
+        self._conv_stem = conv_type(self.in_channels, out_channels, kernel_size=3, stride=stride, bias=False)
+        self._conv_stem_padding = _make_same_padder(self._conv_stem, current_image_size)
+        self._bn0 = batchnorm_type(num_features=out_channels, momentum=bn_mom, eps=bn_eps)
+        current_image_size = _calculate_output_image_size(current_image_size, stride)
+
+        # Build MBConv blocks
+        self._blocks = nn.Sequential()
+        num_blocks = 0
+
+        # Update block input and output filters based on depth multiplier.
+        for idx, block_args in enumerate(self._blocks_args):
+            block_args = block_args._replace(
+                input_filters=_round_filters(block_args.input_filters, width_coefficient, depth_divisor),
+                output_filters=_round_filters(block_args.output_filters, width_coefficient, depth_divisor),
+                num_repeat=_round_repeats(block_args.num_repeat, depth_coefficient),
+            )
+            self._blocks_args[idx] = block_args
+
+            # calculate the total number of blocks - needed for drop_connect estimation
+            num_blocks += block_args.num_repeat
+
+        # Create and add MBConvBlocks to self._blocks
+        idx = 0  # block index counter
+        for block_args in self._blocks_args:
+
+            drop_connect_rate = drop_connect_rate
+            if drop_connect_rate:
+                drop_connect_rate *= float(idx) / num_blocks  # scale drop connect_rate
+
+            # The first block needs to take care of stride and filter size increase.
+            self._blocks.add_module(
+                str(idx),
+                MBConvBlock(
+                    spatial_dims,
+                    block_args.input_filters,
+                    block_args.output_filters,
+                    block_args.kernel_size,
+                    block_args.stride,
+                    current_image_size,
+                    block_args.expand_ratio,
+                    block_args.se_ratio,
+                    block_args.id_skip,
+                    batch_norm_momentum,
+                    batch_norm_epsilon,
+                    drop_connect_rate=drop_connect_rate,
+                ),
+            )
+            idx += 1  # increment blocks index counter
+
+            current_image_size = _calculate_output_image_size(current_image_size, block_args.stride)
+            if block_args.num_repeat > 1:  # modify block_args to keep same output size
+                block_args = block_args._replace(input_filters=block_args.output_filters, stride=[1])
+
+            # Repeat block for num_repeat required
+            for _ in range(block_args.num_repeat - 1):
+                drop_connect_rate = drop_connect_rate
+                if drop_connect_rate:
+                    drop_connect_rate *= float(idx) / num_blocks  # scale drop connect_rate
```

Comment:
codecov says this `drop_connect_rate` is not tested?

Reply:
Thanks! Coincidently I have spotted a logical error here, which would have affected training. This logical error is addressed in next commit. I will think about how to isolate and test `drop_connect_rate`. The `drop_connect` layer itself can certainly be added to unittests, I will add this in next commit.